scripts/read-audit-logs.js

```diff
@@ -0,0 +1,46 @@
+const { XMLParser } = require("fast-xml-parser");
+const ActiveMqHelper = require("../helpers/ActiveMqHelper");
+
+const parser = new XMLParser();
+
+const processNameValuePairs = (nvp) => nvp.nameValuePair;
+
+const processLog = (log) => {
+  const parsed = parser.parse(log);
+  const event = parsed.logEvent ? parsed.logEvent : parsed.auditEvent;
+
+  return {
+    componentId: event.componentId,
+    eventType: event.eventType,
+    receivedAt: event.eventDateTime,
+    nameValuePairs: processNameValuePairs(event.nameValuePairs)
+  };
+};
+
+// eslint-disable-next-line no-nested-ternary
+const dateSort = (a, b) => (a.receivedAt > b.receivedAt ? 1 : b.receivedAt > a.receivedAt ? -1 : 0);
```

Comment:
Couldn't you do `(a, b) => a.receivedAt - b.receivedAt` instead of this nested ternary?

Reply:
It's a string, but if I convert it to a date first then I probably could. Will merge and then fix

Reply:
If it's a string, it's even easier:

```ts
const dateSort = (a, b) => a.localeCompare(b)
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare